File: examples/virt-sandbox.pl
#!/usr/bin/perl

use strict;
use warnings;
use Glib::Object::Introspection;

Glib::Object::Introspection->setup(basename => 'GLib', version => '2.0', package => 'GLib');
Glib::Object::Introspection->setup(basename => 'Gtk', version => '3.0', package => 'Gtk');
Glib::Object::Introspection->setup(basename => 'LibvirtGObject', version => '1.0', package => 'LibvirtGObject');
Glib::Object::Introspection->setup(basename => 'LibvirtSandbox', version => '1.0', package => 'LibvirtSandbox');

LibvirtGObject::init_object_check(undef);

my $cfg = LibvirtSandbox::ConfigInteractive->new("sandbox");
if (int(@ARGV) > 0) {
    $cfg->set_command(@ARGV);
}
if (-t STDIN) {
    $cfg->set_tty(1);
}

my $conn = LibvirtGObject::Connection->new("qemu:///session");
$conn->open(undef);

my $ctxt = LibvirtSandbox::ContextInteractive->new($conn, $cfg);
$ctxt->start();

my $con = $ctxt->get_console();

sub closed {
    Gtk::main_quit();
}

$con->connect("closed", \&closed);
$con->attach_stdio();

Gtk::main();

eval {
    $con->detach();
};
eval {
    $ctxt->stop();
};

File: examples/virt-sandbox.py
#!/usr/bin/python

from gi.repository import LibvirtGObject
from gi.repository import LibvirtSandbox
from gi.repository import GLib
from gi.repository import Gtk
import sys
import os

args = sys.argv[1:]

LibvirtGObject.init_object_check(None)

cfg = LibvirtSandbox.ConfigInteractive.new("sandbox")
if len(args) > 0:
    cfg.set_command(args)
if os.isatty(sys.stdin.fileno()):
    cfg.set_tty(True)

conn = LibvirtGObject.Connection.new("qemu:///session")
conn.open(None)

ctxt = LibvirtSandbox.ContextInteractive.new(conn, cfg)
ctxt.start()

con = ctxt.get_app_console()

def closed(obj, error):
    Gtk.main_quit()

con.connect("closed", closed)
con.attach_stdio()

Gtk.main()

try:
    con.detach()
except:
    pass
try:
    ctxt.stop()
except:
    pass
